Get rid of Group, with the intention that the groups library is used instead

Ericson2314 opened the pull request:
See change log, this does depend on https://github.com/reflex-frp/patch/issues/4 being resolved first, and possibly other issues.

Ericson2314 committed:
Delete group class, the groups package should be used instead

Keep `Additive`, however. See ChangeLog for more details.

The version in the cabal file is bumped to remind whoever does the
future release that this is breaking change.

Closes #4.

Ericson2314 committed:
Bump reflex platform

ali-abrar commented:
Let's see where things actually break without that orphan instance - maybe we can remove reliance on it entirely.

Ericson2314 committed:
Move `Additive` to its own module

Ericson2314 committed:
Fix GHC < 8.6 build

It turns out we were relying on instances from a transitive dependency
on base-orphans. Added that explicitly so that we don't regress on the
instances provided with old GHCs.

Ericson2314 commented on patch.cabal:
```diff
@@ -40,9 +35,13 @@ library
                , transformers >= 0.5.6.0 && < 0.6
                , witherable >= 0.3 && < 0.4
 
+  if impl(ghc < 8.6) -- really, if base < 8.12
+    build-depends: base-orphans >= 0.8 && < 0.9
```
We could get rid of this, and just not provide `:.:` and `:*:` `Additive` instances before GHC 8.6. I don't care which.

Ericson2314 committed:
Merge remote-tracking branch 'origin/develop' into no-group

Ericson2314 committed:
Merge branch 'develop' into no-group

Ericson2314 committed:
Fix typo in change log

Ericson2314 committed:
Merge branch 'module-for-additive' into no-group

Ericson2314 committed:
Merge branch 'module-for-additive' into no-group

Ericson2314 committed:
Fix issue in change log

Ericson2314 committed:
Merge branch 'proper-orphans-dep' into no-group

Ericson2314 committed:
Merge branch 'proper-orphans-dep' into no-group

Ericson2314 committed:
Merge branch 'proper-orphans-dep' into no-group

Ericson2314 committed:
Merge branch 'develop' into no-group

Ericson2314 committed:
Actually do need `semialign` / split-these flag

Ericson2314 committed:
Mention the `split-these` flag is back

Ericson2314 committed:
Merge remote-tracking branch 'origin/develop' into no-group

Ericson2314 committed:
Merge remote-tracking branch 'origin/develop' into no-group

Ericson2314 committed:
Merge branch 'develop' into no-group
